config/initializers/_service_providers.rb

```diff
@@ -2,3 +2,7 @@
                     fetch(Rails.env, {})
 
 VALID_SERVICE_PROVIDERS = JSON.parse(Figaro.env.valid_service_providers)
+
+# merge from dashboard
+require 'feature_management'
+ServiceProviderUpdater.new.run if FeatureManagement.use_dashboard_service_providers?
```

Comment:
I think we should boot the app with an empty SERVICE_PROVIDERS and change this line to a `Thread.new { ServiceProviderUpdater.new.run }`